internal/commands/profile/list.go

```diff
@@ -0,0 +1,54 @@
+package profile
+
+import (
+	"fmt"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cli/user"
+	"github.com/10gen/realm-cli/internal/local"
+	"github.com/10gen/realm-cli/internal/terminal"
+)
+
+const (
+	headerName = "Profile"
+	headerKey  = "API Key"
+)
+
+// CommandMetaList is the command meta for the `profile list` command
+var CommandMetaList = cli.CommandMeta{
+	Use:         "list",
+	Aliases:     []string{"ls"},
+	Display:     "profiles list",
+	Description: "List the profiles of your local CLI environment",
+}
+
+// CommandList is the `profile list` command
+type CommandList struct{}
+
+// Handler is the command handler
+func (cmd *CommandList) Handler(profile *user.Profile, ui terminal.UI, clients cli.Clients) error {
+	profileEnvs, err := local.LoadEnvironments()
+	if err != nil {
+		return err
+	}
+
+	rows := make([]map[string]interface{}, 0)
+	for _, v := range profileEnvs {
+		credentials := "(logged out)"
+		if v.Credentials.PublicAPIKey != "" && v.Credentials.PrivateAPIKey != "" {
+			credentials = fmt.Sprintf("%s (%s)", v.Credentials.PublicAPIKey, v.Credentials.RedactedPrivateAPIKey())
```

Comment:
i'm guessing the `RedactedPrivateAPIKey()` function is what led you to re-use the `user.Credentials` type and add `yaml` tags to it...

we can/should just refactor the `RedactedPrivateAPIKey` logic into a static function (read: not a receiver function on the `user.Credentials` type).  something like `user.RedactAPIKey(value string) string` (and `user.RedactPassword(value string) string`